packages/verkle/src/verkleTree.ts

```diff
@@ -277,10 +278,23 @@ export class VerkleTree {
     let currentDepth = leafNode.depth
 
     while (currentDepth > 0) {
-      const parentKey = currentKey.slice(0, -1)
       const parentIndex = currentKey[currentKey.length - 1]
-      const parentNode = InternalNode.create(currentDepth, this.verkleCrypto)
-      parentNode.children[parentIndex] = currentNode
+      // Parent key should be at least one shorter than the current key length
+      const parentKey = currentKey.slice(0, currentKey.length - 1)
+      const rawNode = await this.get(parentKey)
```

Comment:
I'm not sure that this is a proper use of `this.get`? Shouldn't we use findPath instead for e.g.?

Reply:
Oh, you know, I think i meant to do `trie._db.get` here. Maybe that's the difference I was missing before